programs/fileio.c

```diff
@@ -862,6 +863,205 @@ int FIO_compressFilename(const char* dstFileName, const char* srcFileName,
+                }
+                info->numActualFrames++;
+            }
+            else if (magicNumber == ZSTD_MAGIC_SKIPPABLE_START) {
+                BYTE frameSizeBuffer[4];
+                size_t const readBytes = fread(frameSizeBuffer, 1, 4, srcFile);
+                if (readBytes != 4) {
+                    DISPLAY("There was an error reading skippable frame size");
+                    detectError = 1;
+                    break;
+                }
+                {
+                    U32 const frameSize = MEM_readLE32(frameSizeBuffer);
+                    int const ret = fseek(srcFile, (long)frameSize, SEEK_CUR);
+                    if (ret != 0) {
+                        DISPLAY("Error: could not find end of skippable frame\n");
+                        detectError = 1;
+                        break;
+                    }
+                }
+                info->numSkippableFrames++;
+            }
+            else {
+                detectError = 2;
+                break;
+            }
+        }
+    }
+    fclose(srcFile);
+    return detectError;
+}
+
+static void displayInfo(const char* inFileName, fileInfo_t* info, int displayLevel){
+    double const compressedSizeMB = (double)info->compressedSize/(1 MB);
+    double const decompressedSizeMB = (double)info->decompressedSize/(1 MB);
+    const char* checkString = (info->usesCheck ? "XXH64" : "None");
+    if (displayLevel <= 2) {
+        if (info->canComputeDecompSize) {
+            double const ratio = (info->decompressedSize == 0) ? 0.0 : compressedSizeMB/decompressedSizeMB;
+            DISPLAYOUT("Skippable  Non-Skippable  Compressed  Uncompressed  Ratio  Check  Filename\n");
+            DISPLAYOUT("%9d  %13d  %7.2f MB  %9.2f MB  %5.3f  %s  %s\n",
+                    info->numSkippableFrames, info->numActualFrames, compressedSizeMB, decompressedSizeMB,
+                    ratio, checkString, inFileName);
+        }
+        else {
+            DISPLAYOUT("Skippable  Non-Skippable  Compressed  Check  Filename\n");
+            DISPLAYOUT("%9d  %13d  %7.2f MB  %s  %s\n",
+                    info->numSkippableFrames, info->numActualFrames, compressedSizeMB, checkString, inFileName);
+        }
+    }
+    else{
+        DISPLAYOUT("# Zstandard Frames: %d\n", info->numActualFrames);
+        DISPLAYOUT("# Skippable Frames: %d\n", info->numSkippableFrames);
+        DISPLAYOUT("Compressed Size: %.2f MB (%llu B)\n", compressedSizeMB, info->compressedSize);
+        if (info->canComputeDecompSize) {
+            DISPLAYOUT("Decompressed Size: %.2f MB (%llu B)\n", decompressedSizeMB, info->decompressedSize);
+            DISPLAYOUT("Ratio: %.4f\n", compressedSizeMB/decompressedSizeMB);
+        }
+        if (info->usesCheck) {
+            DISPLAYOUT("Check: XXH64\n");
+        }
+    }
+
+    DISPLAYOUT("\n");
+}
+
+int FIO_listFile(const char* inFileName, int displayLevel){
+    fileInfo_t info;
+
+    /* initialize info to avoid warnings */
```

Comment:
I would recommend to just `memset()` the structure.

As a general rule, try to simplify initialisation to a simple `memset(,0,)`.
It makes much simpler to follow for an external reviewer.
A more complex scheme will require to keep in mind the state of the structure, and track its changes throughout its lifetime. It's more complex, hence lend itself to errors, especially at later stages (when the code is modified again, eventually by someone else).